lib/internal/http2/core.js

```diff
@@ -636,15 +636,21 @@ function initOriginSet(session) {
   if (originSet === undefined) {
     const socket = session[kSocket];
     session[kState].originSet = originSet = new SafeSet();
-    if (socket.servername != null) {
-      let originString = `https://${socket.servername}`;
-      if (socket.remotePort != null)
-        originString += `:${socket.remotePort}`;
-      // We have to ensure that it is a properly serialized
-      // ASCII origin string. The socket.servername might not
-      // be properly ASCII encoded.
-      originSet.add(getURLOrigin(originString));
+    let hostName = socket.servername;
+    if (hostName === null || hostName === false) {
+      if (socket.remoteFamily === 'IPv6') {
+        hostName = `[${socket.remoteAddress}]`;
+      } else {
+        hostName = socket.remoteAddress;
+      }
```

Comment:
> Host: the value sent in Server Name Indication (SNI) ([RFC6066],
      Section 3) converted to lower case; if SNI is not present, the
      remote address of the connection (i.e., the server's IP address)

According to [RFC 8336](https://datatracker.ietf.org/doc/html/rfc8336), when the server name is not set, the server's address should be used instead.